adapters/adtrgtme/adtrgtme.go

```diff
@@ -65,25 +74,24 @@ func (v *adapter) MakeRequests(
 	return requests, errors
 }
 
-func (v *adapter) buildRequestURI(openRTBRequest *openrtb2.BidRequest) (string, error) {
-	if openRTBRequest.Site != nil {
-		if openRTBRequest.Site.ID != "" {
-			return fmt.Sprintf("%s?s=%s&prebid", v.endpoint, openRTBRequest.Site.ID), nil
-		}
-		return "", &errortypes.BadInput{
-			Message: "request.Site.ID is not provided",
-		}
-	} else if openRTBRequest.App != nil {
-		if openRTBRequest.App.ID != "" {
-			return fmt.Sprintf("%s?s=%s&prebid", v.endpoint, openRTBRequest.App.ID), nil
-		}
-		return "", &errortypes.BadInput{
-			Message: "request.App.ID is not provided",
+func (v *adapter) getImpressionExt(imp *openrtb2.Imp) (openrtb_ext.ExtImpAdtrgtme, error) {
```

Comment:
this function doesn't have to be defined as pointer receiver on adapter as this function is not using any of params defined on adapter. So simply - 
``` 
func getImpressionExt(imp *openrtb2.Imp) (openrtb_ext.ExtImpAdtrgtme, error) {
\\
}
```

Reply:
If it is critical for the adapter and it will somehow affect the work. We do it as a separate function.

Reply:
I am not suggesting you to remove this function. 
In `getImpressionExt` function, you are not accessing any value of the adapter struct. I am suggesting to not define it as pointer receiver. Here is the code that you can use -
```
func getImpressionExt(imp *openrtb2.Imp) (openrtb_ext.ExtImpAdtrgtme, error) {
  var bidderExt adapters.ExtImpBidder
	  if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
		  return openrtb_ext.ExtImpAdtrgtme{}, &errortypes.BadInput{
			  Message: "ext.bidder not provided",
		  }
	  }
	  
  var ext openrtb_ext.ExtImpAdtrgtme
	  if err := json.Unmarshal(bidderExt.Bidder, &ext); err != nil {
		  return openrtb_ext.ExtImpAdtrgtme{}, &errortypes.BadInput{
			  Message: "ext.bidder not provided",
		  }
	  }
	  return ext, nil
}
```

Reply:
I do not write that we will remove it. We made it function as you requested.